src/creature_instances.c

```diff
@@ -722,10 +724,74 @@ long instf_first_person_do_imp_task(struct Thing *creatng, long *param)
     TRACE_THING(creatng);
     MapSlabCoord slb_x = subtile_slab_fast(creatng->mappos.x.stl.num);
     MapSlabCoord slb_y = subtile_slab_fast(creatng->mappos.y.stl.num);
+    struct SlabMap* slb;
     if (check_place_to_pretty_excluding(creatng, slb_x, slb_y))
     {
         instf_pretty_path(creatng, NULL);
-    } else
+    }
+    else if (first_person_dig_claim_mode)
```

Comment:
Would it be possible to 'simply' claim the tile/room if there's nothing to dig? So that you would only need to hold the key to capture the room when it would otherwise dig a tunnel?

Of course, when reinforcing you would always need to hold the key.